fvm/environment/unsafe_random_generator.go

```diff
@@ -59,31 +60,47 @@ func (gen ParseRestrictedUnsafeRandomGenerator) UnsafeRandom() (
 func NewUnsafeRandomGenerator(
 	tracer tracing.TracerSpan,
 	blockHeader *flow.Header,
+	txnIndex uint32,
 ) UnsafeRandomGenerator {
 	gen := &unsafeRandomGenerator{
 		tracer:      tracer,
 		blockHeader: blockHeader,
+		txnIndex:    txnIndex,
 	}
 
 	return gen
 }
 
-// This function abstracts building the PRG seed from the entropy source `randomSource`.
-// It does not make assumptions about the quality of the source, nor about
-// its length (the source could be a fingerprint of entity, an ID of an entity,
+// This function abstracts building the PRG seed from the entropy source
+// `randomSource` and `salt`. It does not make assumptions about the quality
+// of the source, nor about its length (the source could be a fingerprint of
+// entity, an ID of an entity, a beacon signature..)
 //
-//	a beacon signature..)
-//
-// It therefore uses a mechansim to extract the source entropy and expand it into
-// the required `seedLen` bytes (this can be a KDF, a MAC, a hash with extended-length output..)
-func seedFromEntropySource(randomSource []byte, seedLen int) ([]byte, error) {
+// It therefore uses a mechansim to extract the source entropy and expand it
+// into the required `seedLen` bytes (this can be a KDF, a MAC, a hash with
+// extended-length output..)
+func seedFromEntropySource(
+	randomSource []byte,
+	salt []byte,
+	seedLen int,
+) (
+	[]byte,
+	error,
+) {
 	// This implementation used HKDF,
 	// but other promitives with the 2 properties above could also be used.
-	hkdf := hkdf.New(func() hash.Hash { return sha256.New() }, randomSource, nil, nil)
+	hkdf := hkdf.New(
+		func() hash.Hash { return sha256.New() },
+		randomSource,
+		salt,
+		nil)
```

Comment:
nit: also not sure what is the right API to use here: should txid go into the sal or info? [rfc5869](https://datatracker.ietf.org/doc/html/rfc5869#section-3.2) is not making it easy.

Reply:
salt seems more appropriate given how info is used (at least based on the wiki "reference" implementation https://en.wikipedia.org/wiki/HKDF)